src/compiler/checker.ts

```diff
@@ -5479,7 +5479,7 @@ namespace ts {
                         specifierCompilerOptions,
                         contextFile,
                         moduleResolverHost,
-                        { importModuleSpecifierPreference: isBundle ? "non-relative" : "relative" },
+                        { importModuleSpecifierPreference: isBundle ? "non-relative" : "relative", importModuleSpecifierEnding: isBundle ? "minimal" : undefined },

```

Comment:
Maybe this is easier to read? But maybe not, I just wanted to write it out for comparison.

```suggestion
                        isBundle ? { importModuleSpecifierPreference: "non-relative", importModuleSpecifierEnding: "minimal" } : { importModuleSpecifierPreference: "relative" },
```

Reply:
Eh, I think I prefer having the single object literal, even if the condition happens to be the same for both fields.